website/content/en/docs/best-practices/resource-pruning.md

```diff
@@ -0,0 +1,149 @@
+		MaxCountSetting: 1,
+	},
+	PreDeleteHook: myhook,
+}
+```
+
+
+| Config Field | Description
+| ------------ | -----------
+| log          | a logger to handle library log messages
+| DryRun       | true means to execute but not to remove resources
+| Clientset    | a client-go Kubernetes ClientSet that will be used for Kube API calls by the library
+| LabelSelector| Kubernetes label selector expression used to find resources to prune
+| Resources    | Kube resource Kinds, currently PodKind and JobKind are supported by the library
+| Namespaces   | a list of Kube Namespaces to search for resources
+| Strategy     | specifies the pruning strategy to execute
+| Strategy.Mode| currently MaxCountStrategy, MaxAgeStrategy, or CustomStrategy are supported
+| Strategy.MaxCountSetting| integer value for *maxcount* strategy, specifies how many resources should remain after pruning executes
+| Strategy.MaxAgeSetting| golang time.Duration string value (e.g. 48h), specifies age of resources to prune
+| Strategy.CustomSettings| a golang map of values that can be passed into a Custom strategy function
+| PreDeleteHook| optionally specifies a golang function to call before pruning a resource
+| CustomStrategy | optionally specifies a golang function that implements a custom pruning strategy
+
+
+### Pruning Execution
+
+Users can invoke the pruning by running the *Execute* function on the pruning configuration
+as follows:
+```golang
+err := cfg.Execute(ctx)
+```
+
+Users might want to implement pruning execution by means of a cron package or simply call the prune
+library based on some other triggering event.
+
+## Pruning Strategies
+
+### maxcount Strategy
+
+A strategy of leaving a finite set of resources is implemented called *maxcount*.  This strategy
+seeks to leave a specific number of resources, sorted by latest, on your cluster.  For example if
+you have 10 resources that would be pruned, and you specified a *maxcount* value of 4, then 6 
+resources would be pruned (removed) from your cluster starting with the oldest resources.
+
+### maxage Strategy
+
+A strategy of removing resources greater than a specific time is called *maxage*.  This strategy
+seeks to remove resources older than a specified *maxage* duration.  For example, a library
+user might specify a value of *48h* to indicate that any resource older than 48 hours would be
+pruned.  Durations are specified using golang's time.Duration formatting (e.g. 48h).
+
+## Pruning Customization
+
+### preDelete Hook
+
+Users can provide a *preDelete* hook when using the [operator-lib prune package][operator-lib-prune].  
+This hook function will be called by the library before removing a resource.  This provides a means to examine
+the resource logs for example, extracting any valued content, before the resource is removed
+from the cluster.
+
+Here is an example of a *preDelete* hook:
+```golang
+func myhook(cfg Config, x ResourceInfo) error {
+	fmt.Println("myhook is called ")
+	return nil
+}
+```
```

Comment:
We should probably provide something that looks at the given attributes. To show what they are and how best to use them. Maybe even go along with your example of extracting valued content.

Reply:
I added a more realistic implementation that pulls out the pod logs to evaluate.